### ISSUE TITLE
Potential Interface Inconsistency between SegmentationLosses and DataParallelCriterion

### ISSUE THREAD

**Issue by huanghoujing** [zhanghang1989/PyTorch-Encoding#165]
First of all, thanks a lot for your excellent code! I found a potential bug in calculating segmentation loss. 

### If not using CUDA

[This line](https://github.com/zhanghang1989/PyTorch-Encoding/blob/ce461dae8d088253dcd9818d2999d4049bce3493/experiments/segmentation/train.py#L75)
```python
self.criterion = DataParallelCriterion(self.criterion).cuda()
```
is not executed, and `self.criterion` is a `SegmentationLosses` object. When it calls the criterion [at this position](https://github.com/zhanghang1989/PyTorch-Encoding/blob/ce461dae8d088253dcd9818d2999d4049bce3493/experiments/segmentation/train.py#L110),
```python
outputs = self.model(image)
loss = self.criterion(outputs, target)
```
wouldn't it cause error [in this case](https://github.com/zhanghang1989/PyTorch-Encoding/blob/ce461dae8d088253dcd9818d2999d4049bce3493/encoding/nn/loss.py#L36)?
```python
def forward(self, *inputs):
    ...
    pred1, se_pred, pred2, target = tuple(inputs)
```
Since it is trying to execute `pred1, se_pred, pred2, target = (outputs, target)`.

### In the case of using CUDA

`self.criterion` is a `DataParallelCriterion` object. It calls `SegmentationLosses` [in a proper way](https://github.com/zhanghang1989/PyTorch-Encoding/blob/ce461dae8d088253dcd9818d2999d4049bce3493/encoding/parallel.py#L160)
```python
output = module(*(input + target), **kwargs)
```

### Conclusion

There is interface inconsistency between `SegmentationLosses` and `DataParallelCriterion`. I think their calling should be different, e.g.
```python
if args.cuda:
    loss = self.criterion(outputs, target)
else:
    loss = self.criterion(*(outputs + target))
```
